tools/distpartitioning/dataset_utils.py

```diff
@@ -147,28 +260,51 @@ def get_dataset(input_dir, graph_name, rank, world_size, num_parts, schema_map):
                 assert (feat_data[constants.STR_FORMAT][constants.STR_NAME]
                 in [constants.STR_NUMPY, constants.STR_PARQUET])
                 # It is guaranteed that num_chunks is always greater 
-                # than num_partitions. 
-                num_chunks = len(feat_data[constants.STR_DATA])
-                read_list = np.array_split(np.arange(num_chunks), num_parts)
-                reader_fmt_meta = {"name": feat_data[constants.STR_FORMAT][constants.STR_NAME]}
+                # than num_partitions.
+                node_data = []
+                num_files = len(feat_data[constants.STR_DATA])
+                if num_files == 0:
+                    continue
+                read_list = np.array_split(np.arange(num_files), world_size)
```

Comment:
Not round-robin, but deterministic if `num_files` and `world_size` is fixed. see example here:
```
>>> a
array([0, 1, 2, 3, 4])
>>> np.array_split(a, 3)
[array([0, 1]), array([2, 3]), array([4])]
```

Reply:
Can you add a unitest for this logic? And if possible, can you provide some APIs under tools/distpartitioning, so users can call these APIs to decide which file to collect/download given the rank and the world_size.
Because our SageMaker solution will rely on this logic.

Reply:
added. pls check here: `//tools/distpartitioning/utils/generate_read_list`